daemon/api_test.go

```diff
@@ -935,3 +939,102 @@ func (s *apiSuite) TestServiceLogs(c *check.C) {
 		Result: []map[string]interface{}{{"message": "hi", "timestamp": "42", "raw": log}},
 	})
 }
+
+func (s *apiSuite) TestMetaIconGet(c *check.C) {
+	// have an “icon” on the system
+	c.Check(os.MkdirAll(dirs.SnapIconsDir, 0755), check.IsNil)
+	c.Check(ioutil.WriteFile(filepath.Join(dirs.SnapIconsDir, "yadda"), []byte("yadda icon"), 0644), check.IsNil)
+
+	s.vars = map[string]string{"icon": "yadda"}
+	req, err := http.NewRequest("GET", "/1.0/icons/yadda", nil)
```

Comment:
I find it slightly curious that we ask the icon by going to /icons, instead of going to <the snap path>/icon.. would be glad to hear the reasoning behind the design.